torch/csrc/jit/runtime/interpreter.cpp

```diff
@@ -34,6 +37,10 @@
 namespace torch {
 namespace jit {
 
+#ifdef USE_DISTRIBUTED
```

Comment:
ifdef for distributed should not appear in interpreter.cpp, this is a sign of strong coupling. I guess an abstraction where we introduce a `SavedThreadLocalState` object that holds all the pytorch-specific thread local state that interpreter needs to save and restore when it suspends. Then we have public functions that the interpreter uses in interpreter continuations:
```
unique_ptr<SavedThreadLocalState> saveThreadLocalState();
restoreThreadLocalState(const SavedThreadLocalState&);
```